servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -147,6 +149,11 @@ public void run() {
         cmdFromMaster = mMasterClient
             .heartbeat(WorkerIdRegistry.getWorkerId(), storeMeta.getUsedBytesOnTiers(),
                 blockReport.getRemovedBlocks(), blockReport.getAddedBlocks());
+        if (WorkerContext.getConf().getBoolean(Constants.IN_TEST_MODE)) {
```

Comment:
I think the clean way is to rewrite `BlockMasterSync` to implement `HeartbeatExecutor`, and then you can use `ScheduledTimer` to control the heartbeat.